zaza/openstack/charm_tests/designate/tests.py

```diff
@@ -255,3 +260,60 @@ class DesignateTests(DesignateAPITests, DesignateCharmTests):
     """Collection of all Designate test classes."""
 
     pass
+
+
+class DesignateBindExpand(BaseDesignateTest):
+    """Test expanding and shrinking bind."""
+
+    TEST_DOMAIN = 'zazabindtesting.com.'
+    TEST_NS1_RECORD = 'ns1.{}'.format(TEST_DOMAIN)
+    TEST_NS2_RECORD = 'ns2.{}'.format(TEST_DOMAIN)
+    TEST_WWW_RECORD = "www.{}".format(TEST_DOMAIN)
+    TEST_RECORD = {TEST_WWW_RECORD: '10.0.0.24'}
+
+    def test_expand_and_contract(self):
+        """Test expanding and shrinking bind."""
+        if not self.post_xenial_queens:
+            raise unittest.SkipTest("Test not supported before Queens")
+
+        domain = designate_utils.create_or_return_zone(
+            self.designate,
+            name=self.TEST_DOMAIN,
+            email="test@zaza.com")
+
+        designate_utils.create_or_return_recordset(
+            self.designate,
+            domain['id'],
+            'www',
+            'A',
+            [self.TEST_RECORD[self.TEST_WWW_RECORD]])
+
+        # Test record is in bind and designate
+        designate_utils.check_dns_entry(
+            self.designate,
+            self.TEST_RECORD[self.TEST_WWW_RECORD],
+            self.TEST_DOMAIN,
+            record_name=self.TEST_WWW_RECORD)
+
+        logging.debug('Adding a designate-bind unit')
+        zaza.model.add_unit('designate-bind')
+        zaza.model.block_until_all_units_idle()
+
+        logging.debug('Performing DNS lookup on all units')
+        designate_utils.check_dns_entry(
+            self.designate,
+            self.TEST_RECORD[self.TEST_WWW_RECORD],
+            self.TEST_DOMAIN,
+            record_name=self.TEST_WWW_RECORD)
+
+        units = zaza.model.get_status().applications['designate-bind']['units']
+        doomed_unit = sorted(units.keys())[0]
+        logging.debug('Removing {}'.format(doomed_unit))
+        zaza.model.destroy_unit('designate-bind', doomed_unit)
+
+        logging.debug('Performing DNS lookup on all units')
+        designate_utils.check_dns_entry(
+            self.designate,
+            self.TEST_RECORD[self.TEST_WWW_RECORD],
+            self.TEST_DOMAIN,
+            record_name=self.TEST_WWW_RECORD)
```

Comment:
Shouldn't this test wait until the model is idle from the destroy_unit? Or does destroy_unit() already block until the unit has gone?